### PR TITLE
Introduce RleVec in chunk data

### PR DESCRIPTION
**Pull request by suprohub**
## Description
A Detailed Description:
it can very cool reduce memory usage, but bit slovy and not recommended to launch in debug.
Its uses RleVec - run lenght encoding vector.
Like xxxxxaaba becomes x5a2b1a1.
Its like a very fast compression.
(log n for index)
Next i added changing world generator in config.
Now you can select superflat or set custom layers.
Also i added /pumpkin dump, for reading what chunks loaded and how it stored.

## Testing
A description of the tests performed to verify the changes:
join to server, see chunks loading
/pumpkin dump and see Rle Vectors
select generator in config file

## Checklist
Things need to be done before this Pull Request can be merged.
- [x] Code optimized
- [x] Skill issues fixed
- [x] Code is well-formatted and adheres to project style guidelines: `cargo fmt`
- [x] Code does not produce any clippy warnings `cargo clippy`

### CODE REVIEW

**Comment by suprohub** [comment]
wha do yoy think?

**Comment by suprohub** [comment]
ok its very slow

**Comment by suprohub** [comment]
memory usage / 1.5, cpu usage * 4 bruuuh

**Comment by suprohub** [comment]
i think its needs optimizations, or else its bad variant

**Comment by suprohub** [comment]
Very big data overhead in crate, needs to write custom implementation

**Comment by suprohub** [comment]
whata fuuu

**Comment by suprohub** [comment]
why it cannot pass lint

**Comment by Snowiiii** [comment]
@suprohub i don't want to be rude, But could you please not just always write a bunch of comments about what currently just comes to you mind. Write Comments which are relevant to the Changes or ask questions when you have any

**Comment by suprohub** [comment]
@Snowiiii

File: pumpkin-world/src/biome.rs [hunk 1]
```diff
@@ -2,13 +2,74 @@ use enum_dispatch::enum_dispatch;
 use serde::{Deserialize, Serialize};
 
 // TODO make this work with the protocol
+// TODO add missing biomes
 // Send by the registry
-#[derive(Serialize, Deserialize, Clone, Copy)]
+#[derive(Serialize, Deserialize, Clone, Copy, Default, Debug)]
 #[non_exhaustive]
 pub enum Biome {
```
Comment (Snowiiii) [inline pumpkin-world/src/biome.rs:9]: This Biome enum is more of an tmp thing. I wanted to parse biome from a JSON.

Comment (suprohub) [reply pumpkin-world/src/biome.rs:9]: Ok. Its needs remove or i can nothing to do?

**Pull request closed without merging**